message/messaging_service.hh

```diff
@@ -342,6 +343,9 @@ public:
     future<> start_listen(locator::shared_token_metadata& stm);
     uint16_t port();
     gms::inet_address listen_address();
+    gms::inet_address broadcast_address() {
+        return _cfg.broadcast_address;
```

Comment:
Messaging service already points to token metadata, as it also needs topology. It could retrieve broadcast address from there as well

Reply:
Nope, apparently we can't rely on _token_metadata early enough, e.g. from:
```
netw::messaging_service::broadcast_address() const at ./message/messaging_service.cc:487
service::retrying_dispatcher::dispatch_to_node(netw::msg_addr, query::forward_request) at ./service/forward_service.cc:270
operator() at ./service/forward_service.cc:602
```

As hit by `topology.test_change_ip`

Reply:
Pity. Not this time then